x-pack/plugins/maps/server/mvt/get_tile.test.ts

```diff
@@ -115,22 +160,78 @@ describe('getGridTile', () => {
   };
 
   test('0.0.0 tile (clusters)', async () => {
-    const tile = await getGridTile(defaultParams);
-    compareTiles('./__tests__/pbf/0_0_0_grid_aspoint.pbf', tile);
+    const pbfTile = await getGridTile(defaultParams);
+    const jsonTile = new VectorTile(new Protobuf(pbfTile));
+    compareJsonTiles(jsonTile, {
+      version: 2,
+      name: 'source_layer',
+      extent: 4096,
+      length: 1,
+      features: [
+        {
+          id: undefined,
+          type: 1,
+          properties: {
+            ['avg_of_TOTAL_AV']: 5398920.390458991,
+            doc_count: 42637,
+          },
+          extent: 4096,
+          pointArrays: [[{ x: 1206, y: 1539 }]],
+        },
+      ],
+    });
   });
 
   test('0.0.0 tile (grids)', async () => {
-    const tile = await getGridTile({ ...defaultParams, requestType: RENDER_AS.GRID });
-    compareTiles('./__tests__/pbf/0_0_0_grid_asgrid.pbf', tile);
+    const pbfTile = await getGridTile({ ...defaultParams, requestType: RENDER_AS.GRID });
+    const jsonTile = new VectorTile(new Protobuf(pbfTile));
+    compareJsonTiles(jsonTile, {
+      version: 2,
+      name: 'source_layer',
+      extent: 4096,
+      length: 1,
+      features: [
+        {
+          id: undefined,
+          type: 3,
+          properties: {
+            ['avg_of_TOTAL_AV']: 5398920.390458991,
+            doc_count: 42637,
+          },
+          extent: 4096,
+          pointArrays: [
+            [
+              { x: 1216, y: 1536 },
+              { x: 1216, y: 1568 },
+              { x: 1184, y: 1568 },
+              { x: 1184, y: 1536 },
+              { x: 1216, y: 1536 },
+            ],
+          ],
+        },
+      ],
+    });
   });
 });
 
-function compareTiles(expectedRelativePath: string, actualTile: Buffer | null) {
-  if (actualTile === null) {
-    throw new Error('Tile should be created');
-  }
-  const expectedPath = path.resolve(__dirname, expectedRelativePath);
-  const expectedBin = fs.readFileSync(expectedPath, 'binary');
-  const expectedTile = Buffer.from(expectedBin, 'binary');
-  expect(expectedTile.equals(actualTile)).toBe(true);
+/**
+ * Verifies JSON-representation of tile-contents
+ * @param actualTileJson
+ * @param expectedLayer
+ */
+function compareJsonTiles(actualTileJson: VectorTile, expectedLayer: ITileLayerJsonExpectation) {
+  const actualLayer: VectorTileLayer = actualTileJson.layers[MVT_SOURCE_LAYER_NAME];
+  expect(actualLayer.version).toEqual(expectedLayer.version);
```

Comment:
Can you just do `expect(actualLayer).toEqual(expectedLayer);` and not need to check each field?

Reply:
discussed offline: VectorTileLayer and VectorTile are class-instances. Explicitly using public API rather than loop over members.